### ISSUE TITLE
3.0 Beta 2 DS/config

### ISSUE THREAD

**Issue by jonmifsud** [symphonists/subsectionmanager#199]
So I've tried to set up Symphony 2.3 - along with SSM3.0beta2. 

The install worked fine - and I could select fields and everything. What's strange is that when I added the datasource.. the result looked somewhat different to what I had expected.

```
  <widgets field-id="32" subsection-id="6" items="1">
    <item id="80" />
  </widgets>
  <widgets field-id="32" subsection-id="6" items="1">
    <item id="80" />
  </widgets>
```

Where I would have expected to be one item - with two fields inside with actual content.

Note I had also pulled the extension from tag 2.3 a couple of days back... but waited for this to be moved into master - as I thought it was unfinished before. (When I pulled & updated the results stayed the same)

**Comment by nilshoerrmann** [comment]
That's quite strange. 
- How many fields did you select from your subsection in your Data Source?
- Is there a `subsectionmanager-storage` file in your `manifest`?

**Comment by jonmifsud** [comment]
Selected two fields - If I select 3 fields I see another `<widgets>` chunk so to say.

yep file exists and this is the content `a:1:{s:6:"fields";a:0:{}}` looks un-populated to me...

**Comment by nilshoerrmann** [comment]
Can you make sure that file is readable and resave you DS, please? Maybe that helps.

**Comment by jonmifsud** [comment]
content stays the same. File gets deleted when I save the DS - and re-created when I visit a page with the DS for the first time.

**Comment by nilshoerrmann** [comment]
Do you use any special field types besides the core ones?

**Comment by nilshoerrmann** [comment]
And could you post the `$dsParamINCLUDEDELEMENTS` array from your DS, please?

**Comment by jonmifsud** [comment]
Not inside the SSM itself no. Just text-input and a text-area.

Below is the includes when I select all the fields inside the DS.

Widgets is the content-area the SSM is linked to.

```
public $dsParamINCLUDEDELEMENTS = array(
    'page',
    'widgets: name',
    'widgets: template-name',
    'widgets: content: formatted',
    'widgets: content: unformatted'
);
```

**Comment by nilshoerrmann** [comment]
Right, thanks. We'll be working on a private project this weekend that uses Symphony 2.3 and Subsection Manager – I'll check if I can recreate and fix your problem then.

**Comment by jonmifsud** [comment]
Ok no problem & no rush. Its a private project too here. If I have some time to spare I'll try have a look as well. Would be in the `field.subsectionmanager.php` probably inside `appendFormattedElement()` ?

**Comment by nilshoerrmann** [comment]
The problem lies most likely in `extension.driver.php` – this is where the extension parses included fields and caches entries.

**Comment by klaftertief** [comment]
Had the same problem and added a fix to pull request #201.

The problem was that with the new abstractions in the 2.3 Datasources, "normal" section DS extend now the `SectionDatasource` class.

**Comment by nilshoerrmann** [comment]
Thanks a lot, Jonas!
